capDL-tool/CapDL/AST.hs

```diff
@@ -92,6 +92,12 @@ data ARMIRQExtraParam =
         target_core :: Word }
     deriving (Show, Eq, Ord, Typeable, Data)
 
+data ARMSGISignalExtraParam =
+    ARMSGISignalIrq {
+        sgi_irq :: Word }
+    deriving (Show, Eq, Ord, Typeable, Data)
```

Comment:
I'm not sure if it's worth changing but I don't see why this is an `ExtraParam` instead of being a normal `ObjParam`. From memory, the extra params were originally added for specific parameters that were needed for initialising but that otherwise would have been abstracted away by the CapDL language as not relevant for security analysis. The irq being signalled and the signal's target feels to me like something that should be 'first-class' parameters, particularly since they are also represented in seL4's abstract specification.

Reply:
Another reason they may have been extra is because Arm has `seL4_IRQControl_GetTriggerCore` in addition to the generic `seL4_IRQControl_Get` to set some extra optional parameters, so it are really extra parameters that are sometimes set explicitly and sometimes just use the default values, depending on the syscall used.